crates/core/src/client/helpers.rs

```diff
@@ -101,7 +100,7 @@ pub fn decode_eth_call_return(call_result: &[FieldElement]) -> Result<Vec<FeltOr
     // fail is Kakarot API changes)
     // Declare Vec of Result
     let mut segmented_result: Vec<FeltOrFeltArray> = Vec::new();
-    let mut tmp_array_len: FieldElement = *call_result.get(0).ok_or_else(|| {
+    let mut tmp_array_len: FieldElement = *call_result.first().ok_or_else(|| {
         EthApiError::OtherError(anyhow::anyhow!("Cannot parse and decode return arguments of Kakarot call",))
```

Comment:
I guess we can create an issue for these error types? Like the issue would be 
replace OtherError occurrences by meaningful errors?

task lists would be 
grep all OtherError occurences
replace by appropriate error